### PR TITLE
feat: completed the header component

### PR DESCRIPTION
**Pull request by hokagedemehin**
## Description
This pull request completes the header component of the web application which follows the new design patterns

## Related Issue
#34 

## Screenshots (if appropriate):
image

image

### CODE REVIEW

File: src/components/molecules/main-nav/main-nav.tsx [hunk 1]
```diff
@@ -5,47 +5,50 @@ import {
...
+  useDisclosure,
+  Button
 } from "@chakra-ui/react";
 import { Fragment } from "react";
 import NavLink from "components/atoms/nav-link/nav-link";
 import Icon from "components/atoms/icon/icon";
 import { FiMenu, FiExternalLink } from "react-icons/fi";
 import { useRouter } from "next/router";
-
+import DefaultButton from "../../atoms/button/button";
```
Comment (TAKANOME-DEV) [inline src/components/molecules/main-nav/main-nav.tsx:16]:
```suggestion
import DefaultButton from "components/atoms/button/button";
```

File: src/components/molecules/main-nav/main-nav.tsx [hunk 2]
```diff
@@ -5,47 +5,50 @@ import {
...
 
-const MainNav = ({ navList }: MainNavProps) => {
+const MainNav = ({ navList, mobileList }: MainNavProps) => {
   const { isOpen, onOpen, onClose } = useDisclosure();
   const router = useRouter();
 
   return (
     <nav aria-label="Main">
-      <ul className="hidden gap-4 lg:flex">
+      <ul className="lg:flex space-x-8 hidden items-center">
```
Comment (TAKANOME-DEV) [inline src/components/molecules/main-nav/main-nav.tsx:28]:
```suggestion
      <ul className="sm:flex space-x-8 hidden items-center">
```

File: src/components/molecules/main-nav/main-nav.tsx [hunk 3]
```diff
@@ -5,47 +5,50 @@ import {
...
+          className=""
+        >
+          <span className="text-white">Contact Us</span>
+        </DefaultButton>
       </ul>
       <div
         aria-label="Menu"
         aria-expanded="false"
-        className="flex h-[70px] items-center lg:hidden"
+        className="lg:hidden h-[70px] flex items-center"
```
Comment (TAKANOME-DEV) [inline src/components/molecules/main-nav/main-nav.tsx:51]:
```suggestion
        className="sm:hidden h-[70px] flex items-center"
```

**Review by TAKANOME-DEV — CHANGES_REQUESTED** [review]
Thanks for your contribution @hokagedemehin. Looking great :+1: 

Just a couple of suggestions:
- let's only show the mobile nav at `sm` instead of `lg`
- I think using absolute paths (components/atoms) is better than relative paths (../../atoms)

File: src/components/molecules/main-nav/main-nav.tsx [hunk 4]
```diff
@@ -5,47 +5,50 @@ import {
...
   const router = useRouter();
 
   return (
     <nav aria-label="Main">
-      <ul className="hidden gap-4 lg:flex">
+      <ul className="lg:flex space-x-8 hidden items-center">
         {navList.map((navItem, index) => (
           <Fragment key={index}>
-            <li className="flex items-center gap-2">
+            <li>
```
Comment (chadstewart) [inline src/components/molecules/main-nav/main-nav.tsx:31]:
```suggestion
            <li className="flex items-center gap-2>
```
Please replace this className

File: src/components/molecules/main-nav/main-nav.tsx [hunk 5]
```diff
@@ -5,47 +5,50 @@ import {
...
                 activeLink={router.pathname === navItem.url}
                 {...navItem}
               />
-              {navItem.externalLink && (
-                <Icon
-                  Icon={FiExternalLink}
-                  size={22}
-                  iconAlt="external link for TechIsHiring newsletter"
-                />
-              )}
```
Comment (chadstewart) [inline src/components/molecules/main-nav/main-nav.tsx:40]:
```
{navItem.externalLink && (
  <Icon
    Icon={FiExternalLink}
    size={22}
    iconAlt="external link for TechIsHiring newsletter"
 />
)}
```

Please replace this code block. This is there for accessibility.

File: src/components/molecules/main-nav/main-nav.tsx [hunk 6]
```diff
@@ -57,12 +60,10 @@ const MainNav = ({ navList }: MainNavProps) => {
 
             <DrawerBody>
               <ul>
-                {navList.map((navItem, index) => (
+                {mobileList.map((navItem, index) => (
```
Comment (chadstewart) [inline src/components/molecules/main-nav/main-nav.tsx:63]:
```suggestion
                {navList.map((navItem, index) => (
```
Please remove this instance of mobileList.

File: src/components/organisms/header/header.tsx [hunk 7]
```diff
@@ -2,24 +2,18 @@ import Logo from "components/atoms/logo/logo";
 import HeaderText from "components/atoms/typography/heading-text";
 import MainNav from "components/molecules/main-nav/main-nav";
 import useMainNav from "lib/hooks/use-nav";
+import useMobileNav from "lib/hooks/use-mobile-nav";
```
Comment (chadstewart) [inline src/components/organisms/header/header.tsx:5]:
```suggestion
```
Please remove this import of mobileList.

File: src/components/organisms/header/header.tsx [hunk 8]
```diff
@@ -2,24 +2,18 @@ import Logo from "components/atoms/logo/logo";
...
 
 const Header = () => {
   const navList = useMainNav();
+  const mobileList = useMobileNav();
```
Comment (chadstewart) [inline src/components/organisms/header/header.tsx:9]:
```suggestion
```
Please remove this instance of mobileList.

File: src/components/organisms/header/header.tsx [hunk 9]
```diff
@@ -2,24 +2,18 @@ import Logo from "components/atoms/logo/logo";
...
-          className="hidden w-[440px] px-5 md:block"
-          level="h1"
-          size="md"
-        >
-          &apos;Where great jobs and great Engineers go to find each
-          other!&apos;
-        </HeaderText>
       </div>
-      <MainNav navList={navList} />
+      <MainNav navList={navList} mobileList={mobileList} />
```
Comment (chadstewart) [inline src/components/organisms/header/header.tsx:16]:
```suggestion
      <MainNav navList={navList} />
```
Please remove this instance of mobileList.

File: src/lib/hooks/use-mobile-nav.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    {
+      url: "/contact",
+      text: "Contact Us"
+    }
+  ];
+
+  return navList;
+};
+
+export default useMobileNav;
```
Comment (chadstewart) [inline src/lib/hooks/use-mobile-nav.tsx:24]: Please delete this file. The useNav hook should be the only hook for the nav. There is no need to make another hook only for the mobile nav.

Also, could you add the contact us object to the useNav hook please?

**Review by chadstewart — CHANGES_REQUESTED** [review]
Thanks so much for your contribution! Really appreciate the work that you've put in and I especially liked the idea of making the logo just text since honestly that's what it is currently.

Just wanted to request a few changes mainly around the mobileList hook you created. That's not needed and I really only want useNav to be where the navigation lives. Having to maintain two hooks for navigation will lead to someone forgetting to add an entry in either.

Again, thanks for your contribution!

File: src/components/molecules/main-nav/main-nav.tsx (hunk 5 above)
Comment (hokagedemehin) [reply src/components/molecules/main-nav/main-nav.tsx:40]: please what do you mean by replacing this code block?

Comment (chadstewart) [reply src/components/molecules/main-nav/main-nav.tsx:40]: You deleted it in your PR. Could you please put it back?

File: src/components/molecules/main-nav/main-nav.tsx [hunk 11]
```diff
@@ -60,18 +70,18 @@ const MainNav = ({ navList }: MainNavProps) => {
...
                     }`}
                   >
-                    <NavLink {...navItem} />
                     {navItem.externalLink && (
                       <Icon
                         Icon={FiExternalLink}
                         size={22}
                         iconAlt="external link for TechIsHiring newsletter"
                       />
                     )}
```
Comment (hokagedemehin) [inline src/components/molecules/main-nav/main-nav.tsx:83]: @chadstewart This is the code block you are asking for. I never deleted it, it is only visible in the drawer component

**Comment by hokagedemehin** [comment]
I made a little adjustment to the homepage test.

please check it out to let me know what you think.

it is 3:30 am over here 😃

File: src/components/molecules/main-nav/main-nav.tsx (hunk 11 above)
Comment (hokagedemehin) [reply src/components/molecules/main-nav/main-nav.tsx:90]: This is not the code block you are talking about so I will close this conversation

**Review by chadstewart — APPROVED** [review]
Looked over everything and it looks great! Great job and thanks so very much for your contributions!

**Pull request merged**